iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+
+
+    forward_list<CTorTest, TESTLISTCAPACITY>::const_iterator iter_sut1 = cut1.cbefore_begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::const_iterator iter_sut2 = cut1.cbefore_begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::iterator iter_sut3 = cut1.begin();
+    forward_list<CTorTest, TESTLISTCAPACITY>::iterator iter_sut4 = cut1.end();
+
+    ASSERT_THAT(iter_sut1 == iter_sut3, Eq(false));
+    ASSERT_THAT(iter_sut3 == iter_sut1, Eq(false));
+
+    ASSERT_THAT(iter_sut1 == iter_sut2, Eq(true));
+    ASSERT_THAT(iter_sut4 == iter_sut3, Eq(false));
+}
+
+
+TEST_F(forward_list_test, IteratorTraitsGetValueType)
+{
+    forward_list<int, 10> cut;
+
+    cut.emplace_front(5);
+    auto iter{cut.begin()};
+
+    // using a function call here is closer to the actual use case (-> intentionally did not inline all code here)
+    auto ret = iteratorTraitReturnDoubleValue(iter);
+
+    EXPECT_THAT(ret, Eq(10));
+}
+
+TEST_F(forward_list_test, IteratorTraitsCheckIteratorCategoryOnConstIterator)
+{
+    int cnt = 0;
+    auto iter = sut.cbefore_begin();
+    if (typeid(std::iterator_traits<decltype(iter)>::iterator_category) == typeid(std::random_access_iterator_tag))
+    {
+        ASSERT_TRUE(1);
+    }
+    else if (typeid(std::iterator_traits<decltype(iter)>::iterator_category) == typeid(std::forward_iterator_tag))
+    {
+        cnt = 1;
+    }
+
+    EXPECT_THAT(cnt, Eq(1));
+}
+
+TEST_F(forward_list_test, EmptyAfterClear)
+{
+    sut.emplace_front(5);
+    sut.clear();
+    EXPECT_THAT(sut.empty(), Eq(true));
+}
+
+TEST_F(forward_list_test, SizeZeroAfterClear)
+{
+    sut.emplace_front(5);
+    sut.clear();
+    EXPECT_THAT(sut.size(), Eq(0));
+}
+
+TEST_F(forward_list_test, CopyConstructor)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1;
+    cut1.emplace_front(101);
+    cut1.emplace_front(102);
+    EXPECT_THAT(customCTor, Eq(2));
+
+    forward_list<CTorTest, TESTLISTCAPACITY> cut2(cut1);
+
+    EXPECT_THAT(customCTor, Eq(2));
+    EXPECT_THAT(copyCTor, Eq(2));
+    EXPECT_THAT(moveCTor, Eq(0));
+    EXPECT_THAT(moveAssignment, Eq(0));
+    EXPECT_THAT(copyAssignment, Eq(0));
+    auto iter = cut2.begin();
+    EXPECT_THAT(iter->m_value, Eq(102));
+    ++iter;
+    EXPECT_THAT(iter->m_value, Eq(101));
+    EXPECT_THAT(cut2.empty(), Eq(false));
+    EXPECT_THAT(cut2.size(), Eq(2));
+}
+
+TEST_F(forward_list_test, CopyConstructorWithEmptyForwardList)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1;
+    forward_list<CTorTest, TESTLISTCAPACITY> cut2(cut1);
+    EXPECT_THAT(copyCTor, Eq(0));
+    EXPECT_THAT(cut2.size(), Eq(0));
+    EXPECT_THAT(cut2.empty(), Eq(true));
+}
+
+TEST_F(forward_list_test, CopyConstructorWithFullForwardList)
+{
+    forward_list<CTorTest, TESTLISTCAPACITY> cut1;
+    decltype(CTorTest::m_value) i = 0;
+
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        cut1.emplace_front(i);
+    }
+
+    forward_list<CTorTest, TESTLISTCAPACITY> cut2(cut1);
+    for (auto& fl : cut2)
```

Comment:
I still find the fl confusing. I call those element in general when the type of the container elements is unspecific or does not matter. Not important but for clarity (with fl I always need to think what is meant, forward list? ... but this would be wrong since it is an reference to an element of such a list, this breaks the flow when reading).